.github/workflows/server-canary-deployment.yml

```diff
@@ -0,0 +1,77 @@
+name: Server Canary Deployment
+
+permissions:
+  contents: read
+
+on:
+  workflow_dispatch:
+    inputs:
+      commit_sha:
+        description: The commit SHA to deploy. When not passed, it deploys the latest commit available in `main`.
+        required: false
+
+concurrency:
+  group: deployment-canary-${{ github.head_ref }}
+  cancel-in-progress: true
+
+env:
+  FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
```

Comment:
This one is not defined in the GH Actions secrets. Since you can deploy with this token, we should make use the environment secret groups, so that only PRs applied by a Tuist employee can be actually deployed, so this doesn't end up being a security risk.

Reply:
I scoped the secret to each environments.